recipes/_start.rb

```diff
@@ -5,11 +5,30 @@
 #
 # Ensures the splunk instance is running, and performs post-start tasks.
 
+ulimit_command = "ulimit -n #{node['splunk']['limits']['open_files']}"
+init_file_path = '/etc/init.d/splunk'
+restart_flag = !(File.exist?(init_file_path) && File.readlines(init_file_path).grep(/#{ulimit_command}/).any?)
+
 # We want to always ensure that the boot-start script is in place on non-windows platforms
 execute "#{node['splunk']['cmd']} enable boot-start -user #{node['splunk']['user']}" do
   not_if { platform_family?('windows') }
 end
 
+ruby_block 'insert ulimit' do
+  block do
+    file = Chef::Util::FileEdit.new(init_file_path)
+    file.insert_line_after_match(/^RETVAL=\d$/, ulimit_command)
+    file.write_file
+  end
+  not_if { platform_family?('windows') }
+end
+
+ruby_block 'restart-splunk' do
+  block { true }
+  notifies :touch, 'file[splunk-marker]', :immediately
+  only_if { restart_flag }
```

Comment:
This guard needs adjustment as well, because windows

We want to notify for restart:
not_if (windows OR ( init file exists AND correct ulimit setting is present prior to editing))